src/abstracts/SablierV2MerkleStreamer.sol

```diff
@@ -107,6 +118,15 @@ abstract contract SablierV2MerkleStreamer is
 
     /// @dev Validates the parameters of the `claim` function, which is implemented by child contracts.
     function _checkClaim(uint256 index, bytes32 leaf, bytes32[] calldata merkleProof) internal view {
+        // Safe Interactions: query the protocol fee. This is safe because it's a known Sablier contract that does
+        // not call other unknown contracts.
+        UD60x18 protocolFee = LOCKUP.comptroller().protocolFees(ASSET);
+
+        // Checks: the protocol fee is zero.
+        if (protocolFee.gt(ud(0))) {
+            revert Errors.SablierV2MerkleStreamer_ProtocolFeeNotZero();
+        }
+
```

Comment:
It would be better to perform the expiration check first.

- There will be many many more expired campaigns that there will be non-zero protocol fees, so making the check sooner saves gas.
- As a rule of thumb, Checks should come before Interactions (even when they are safe).

Reply:
> It would be better to perform the expiration check first.

nope, the protocol fee check has the highest priority. if there is a campaign with no expiration, this check is unnecessary, but for the protocol fee, it is always a concern if it is non-zero.

> There will be many many more expired campaigns that there will be non-zero protocol fees, so making the check sooner saves gas.

this would affect the gas only if the transaction is going to revert,  should we focus on this specific case (which is not going to happen through our UI), or on the logical aspect? it makes a lot more sense to use the protocol fee as a wall at the entrance of the function

> As a rule of thumb, Checks should come before Interactions (even when they are safe)

ya, agree, but how is it related here since we need to perform an interaction in order to make the check? no matter what you do CEI is not respected here (no grey area)

Reply:
> the protocol fee check has the highest priority
> for the protocol fee, it is always a concern if it is non-zero

I don't see why it has a higher priority. They are both reverts. What matters is what revert is _more likely_ to occur.

> this would affect the gas only if the transaction is going to revert, should we focus on this specific case (which is not going to happen through our UI), or on the logical aspect?

Yes, we should focus on reverts in this case, because we should expect there to be quite a few reverted txs (not due to the protocol fee check):

1. We may make errors in how we generate the Merkle proofs
2. Other airdrop-claiming UIs (such as Earnifi) might make errors

As I've said in my previous reply, it is more likely that one of the other reverts will be triggered than the protocol fee.

>  how is this related here since we need to perform an interaction in order to make the check? no matter what you do CEI is not respected here

Making the external call after _some other checks_ is better than making it before _all checks_.

While writing this response, I realized that it would be better to move the protocol fee check at the end of the `_checkClaim` function.

Reply:
I pushed a commit let me know what you think @andreivladbrg

Reply:
> I pushed a commit let me know what you think

i still don't agree with you here, the main reason being that it is a check that is not in control of either the claim caller or the campaign admin - this is why i say it has the higher priority.

but being a small thing, i am ok with keeping it as you want